djangocms_versioning/management/commands/create_versions.py

```diff
@@ -0,0 +1,96 @@
+            choices=[key for key, value in constants.VERSION_STATES],
+            help=f"state of newly created version object (defaults to {constants.DRAFT})"
+        )
+        parser.add_argument(
+            "--username",
+            type=str,
+            help="Username of user to create the missing Version objects"
+        )
+        parser.add_argument(
+            "--userid",
+            type=int,
+            help="User id of user to create the missing Version objects"
+        )
+
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            help="Do not change the database",
+        )
+
+    def get_user(self, options):
+        if DEFAULT_USER is not None:
+            try:
+                return User.objects.get(pk=DEFAULT_USER)
+            except User.DoesNotExist:
+                raise CommandError(f"No user with id {DEFAULT_USER} found "
+                                   f"(specified as DJANGOCMS_VERSIONING_DEFAULT USER in settings.py")
+
+        if options["userid"] and options["username"]:
+            raise CommandError("Only either one of the options '--userid' or '--username' may be given")
+        if options["userid"]:
+            try:
+                return User.objects.get(pk=options["userid"])
+            except User.DoesNotExist:
+                raise CommandError(f"No user with id {options['userid']} found")
+        if options["username"]:
+            try:
+                return User.objects.get(**{USERNAME_FIELD: options["username"]})
+            except User.DoesNotExist:
+                raise CommandError(f"No user with name {options['username']} found")
+        return None
+
+    def handle(self, *args, **options):
+        user = self.get_user(options)
+
+        for versionable in _cms_extension().versionables:
+            Model = versionable.content_model
+            content_type = ContentType.objects.get_for_model(Model)
+            version_ids = Version.objects.filter(content_type_id=content_type.pk).values_list("object_id", flat=True)
+            unversioned = Model.admin_manager.exclude(pk__in=version_ids).order_by("-pk")
+            self.stdout.write(self.style.NOTICE(
+                f"{len(version_ids) + len(unversioned)} objects of type {Model.__name__}, thereof "
+                f"{len(unversioned)} missing Version object"
+            ))
+            if not options["dry_run"]:
+                if user is None:
+                    raise CommandError("Please specify a user which missing Version objects shall belong to "
+                                       "either with the DJANGOCMS_VERSIONING_DEFAULT_USER setting or using "
+                                       "command line arguments")
+                for orphan in unversioned:
+                    try:
+                        Version.objects.create(
+                            content=orphan,
+                            state=options["state"],
```

Comment:
I don't think that the state should be set, I think this needs to be calculated. The reason I think this is that you could have a grouper object (Page) with many content objects (PageContent), if you blindly set them all to draft you'll get an error where you can only have one draft, it could be that the state machine handles this and forces the existing version to archived, but it could be that the content objects are not in the correct order. 

There's some additional complexities when filtering to find what other versions a grouper has because it can have grouping filters such as language.

Reply:
What do you think?

Reply:
To be fair, that would be a fair assumption what you have proposed. I was thinking more a corrupt setup. 

If there ever as such an occurrence I guess we could point to documentation in the code that this solution is based on that assumption / scenario that you propose. There will be a day where someone comes with such a corrupt setup, maybe they manually deleted Version objects or something (I once did it trying to remove a package, this could be improved by fixing the deletion where you can delete things, a separate issue that I always wanted the time to fix, needs to simply change the source version). My biggest fear of getting more people involved with V4 was just that, corrupt Version history. Maybe fix it when someone comes with it. I had done something similar although it doesn't handle other contenttypes i.e. Page / Pagecontent only, that was in the 3 to 4 migration package where the version states were kept.

Reply:
I think I'm almost there. But the added complexity will require some tests. Will push an update later. :-)

Reply:
@Aiky30 I've updated the PR. It will successfully recover missing Version objects even if there are multiple content objects per grouper. It also reflects additional grouping fields. A test is added that runs on models with (Poll) and without (Blog)  additional grouping field.